draft-ietf-regext-rdap-geofeed.md

```diff
@@ -112,8 +108,8 @@ Registry (see (#structured_syntax_suffixes_registry)).
 
 The "type" JSON value in a Geofeed link object SHOULD be set to the "application/geofeed+csv" media type.
 
-If alternative geofeed formats (beside CSV) are defined in the future, they could be included in future versions of
-this specification.
+If alternative geofeed formats beside CSV are defined in the future, they could be included in future versions of this
+specification.
```

Comment:
This text is also the same in this PR, but is this saying that content types other than application/geofeed+csv will not be used by servers that comply with this specification unless this specification is updated to explicitly permit their use?  If that's the case, maybe an option here is to permit the use of other registered content types for this sort of information, while noting that the main use case here will be geofeed files.

Reply:
> This text is also the same in this PR, but is this saying that content types other than application/geofeed+csv will not be used by servers that comply with this specification unless this specification is updated to explicitly permit their use?

Precisely, to help keep this spec aligned with RFC 9092 and its update.

> If that's the case, maybe an option here is to permit the use of other registered content types for this sort of information, while noting that the main use case here will be geofeed files.

Wouldn't that be too permissive? Could you please propose some text to help clarify your suggestion? :)

Reply:
> Wouldn't that be too permissive? Could you please propose some text to help clarify your suggestion? :)

This points toward a more general question about link-based extensions.  If the relevant link relation and media type were already registered, then geofeed link objects could be validly included in an RDAP response today, without any additional extension work.  The extension identifier therefore has to have some additional significance past "a link with relation R and type T may appear in responses from this server".  For this document, that could be something like (new section 2.1):

```
2.1.  Extension

   A new RDAP extension "geofeed1" will be returned by servers that host
   Geofeed data.  A server that includes this extension identifier in its
   responses is indicating that it will return Geofeed data in IP network
   objects in accordance with this document, where that data is available.
```

The "alternative geofeed formats" text would then become:

```
A server MAY return link objects with the "geo" link relation with types
other than "application/geofeed+csv", even when the server includes the
"geofeed1" extension identifier in its responses.
```

What do you think?

Reply:
TBH, I think what we have presently in section 2.1 and the "alternative geofeed formats" sentence is necessary but perhaps not sufficient. Please allow me to explain.

Section 2.1 as-is focuses on the "new link object for the geofeed file URL", thus aligning with the concept from RFC 9092 and its update. Further, it provides a way for "accessing the geofeed data through RDAP" by providing geofeed file URL info; afraid "return Geofeed data in IP network objects" per the new text might be misconstrued to include structured geofeed data (prefix, location fields) within RDAP responses. IMHO, the current text in section 2.1 seems more tailored.

> A server MAY return link objects with the "geo" link relation with types
other than "application/geofeed+csv", even when the server includes the
"geofeed1" extension identifier in its responses.

This is an interesting statement. Let's test-drive this. Say, I define a new media type "application/long-lat-for-my-home+json" that has nothing to do with the geofeed concept, I think the "geofeed1" extension anyway cannot concern itself with such a new web link and that should be ok because of the allowance from RFC 9083 section 4.2. But, if a new "application/geofeed+json" media type, say, gets defined, would expect the extension to get bumped to "geofeed2" per the current "alternative geofeed formats" sentence because this RDAP extension concerns with geofeed formats.

How about we keep the current "If alternative geofeed formats beside CSV are defined in the future, they could be included in future versions of this specification." sentence (where "versions" connotes "geofeedN" extension) but also follow it up with the above sentence for other "geo" links with non-geofeed types? That should help green-flag an earlier comment in the WG about using "geo" rel for other, non-geofeed links.

We can push this out and see what the WG thinks. :)

Reply:
> Section 2.1 as-is focuses on the "new link object for the geofeed file URL", thus aligning with the concept from RFC 9092 and its update. Further, it provides a way for "accessing the geofeed data through RDAP" by providing geofeed file URL info; afraid "return Geofeed data in IP network objects" per the new text might be misconstrued to include structured geofeed data (prefix, location fields) within RDAP responses. IMHO, the current text in section 2.1 seems more tailored.

Using "accessing ..." instead of "return ..." is fine, but the key thing is that the presence of the identifier indicates that the server supports hosting this data and will return it if it's available, so I think including some text along those lines would be good.

> This is an interesting statement. Let's test-drive this. Say, I define a new media type "application/long-lat-for-my-home+json" that has nothing to do with the geofeed concept, I think the "geofeed1" extension anyway cannot concern itself with such a new web link and that should be ok because of the allowance from RFC 9083 section 4.2.

This sounds fine to me.

> But, if a new "application/geofeed+json" media type, say, gets defined, would expect the extension to get bumped to "geofeed2" per the current "alternative geofeed formats" sentence because this RDAP extension concerns with geofeed formats.

That may well happen, but if that new media type were registered by way of some other document, it would be open to servers to start returning those links immediately.  The only reason to register 'geofeed2' (or 'geofeedjson' or similar) in that case is so that the server can communicate to clients that it hosts that data and will return it if it has it.

Another way of putting this is that there are two things I'm concerned about:

 - clients assuming that any link with the relation "geo" will contain geofeed data; and
 - servers/clients interpreting the current text as implicitly prohibiting the use of new geo/geofeed-related media types in RDAP except by way of new extensions.

So another way of phrasing the last part might be:

```
A server that includes the "geofeed1" extension identifier in its responses is indicating that it hosts geofeed data and will provide access to that data by way of its responses, in accordance with the text of this document.  However, this document does not restrict the use of other media types that relate to geographic information, or the use of link objects more generally.  

Although a server may use registered media types in its link objects without any restrictions, it may be useful to define new RDAP extensions for those media types in order for the server to communicate to clients that it hosts data using that type and will make it available, in the same way that the server does with the "geofeed1" extension identifier.
```

Reply:
> Only change I would suggest is to "indicating that it hosts geofeed data" since an RDAP server returning a response is just pointing to a geofeed file resource which is most likely hosted at a network operator site. In fact, we should change the example in section 2.4 to have "example.com" in "href" to contrast with "example.net" in "value". :)

Yep, these are both good points.

> If ok, we could replace the current "alternative geofeed formats" sentence with above text. That way we end up keeping more-tailored other text while explaining extensibility for such links. Thoughts?

This updated text looks good to me, thanks.